ui/src/app/machines/views/MachineDetails/MachineStorage/AvailableStorageTable/BulkActions/CreateDatastore/CreateDatastore.tsx

```diff
@@ -38,21 +38,12 @@ type Props = {
 
 const getInitialName = (disks: Disk[]) => {
   if (!disks || disks.length === 0) {
-    return "datastore0";
+    return "datastore1";
   }
-  const datastoresCount = disks.reduce<number>((count, disk) => {
-    if (isDatastore(disk.filesystem)) {
-      count += 1;
-    }
-    if (disk.partitions?.length) {
-      disk.partitions.forEach((partition) => {
-        if (isDatastore(partition.filesystem)) {
-          count += 1;
-        }
-      });
-    }
-    return count;
-  }, 0);
+  const datastoresCount = disks.reduce<number>(
+    (count, disk) => (isDatastore(disk.filesystem) ? count + 1 : count),
```

Comment:
Driveby fix. We actually only want to count the disks, not the partitions. Also the api starts at 1, so I've updated it to start at 1 here too

Reply:
Nice. Might be good to leave a comment here about starting at 1 like you did in the test.

Reply:
Ah, thought I did already. Adding now